File: Updog.Client/src/user/domain/user-credentials.ts
/**
 * Username password combo to log in a user.
 */
export class UserCredentials {
    /**
     * Create a new set of user credentials.
     * @param username The username to auth under.
     * @param password The password to auth with.
     */
    constructor(public username: string, public password: string) {}
}

File: Updog.Domain/User/UserCredentials.cs
namespace Updog.Domain {
    public class UserCredentials {
        #region Properties
        public string Username { get; }

        public string Password { get; }
        #endregion

        #region Constructor(s)
        public UserCredentials(string username, string password) {
            Username = username;
            Password = password;
        }
        #endregion
    }
}